src/pages/ocpDetails/priceListModal.tsx

```diff
@@ -30,11 +30,21 @@ interface Props extends InjectedTranslateProps {
 }
 
 class PriceListModalBase extends React.Component<Props> {
-  public componentDidUpdate() {
-    const { fetch, isOpen, providers, priceListStatus: status } = this.props;
-    if (isOpen && status === FetchStatus.none) {
+  public shouldComponentUpdate(nextProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      return true;
+    }
+    if (nextProps.priceListStatus !== this.props.priceListStatus) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
A word of caution using shouldComponentUpdate. If anything is updated (e.g., if we later add inline editing, pagination, etc.), it will need to be checked here or componentDidUpdate/render won't be called as expected. I found this to be problematic elsewhere in the koku UI and since removed it.

Reply:
I can remove it. Don't know why I used it in the first place.